src/migration/MigrationExecutor.ts

```diff
@@ -284,14 +384,18 @@ export class MigrationExecutor {
             const migrationTimestamp = parseInt(migrationClassName.substr(-13));
             if (!migrationTimestamp)
                 throw new Error(`${migrationClassName} migration name is wrong. Migration class name should have a JavaScript timestamp appended.`);
-
-            return new Migration(undefined, migrationTimestamp, migrationClassName, migration);
+            const hash = this.calculateHash(migration);
+            return new Migration(undefined, migrationTimestamp, migrationClassName, hash, migration);
         });
 
         // sort them by timestamp
         return migrations.sort((a, b) => a.timestamp - b.timestamp);
     }
 
+    protected calculateHash(migration: MigrationInterface): string {
+        return sha1(migration);
```

Comment:
What about use `RandomGenerator.sha1`

Reply:
What library is `RandomGenerator.sha1`? `object-hash` looks like reasonable choice. Popular on http://npmjs.com and support node & browser.

Reply:
it is typeorm util
https://github.com/typeorm/typeorm/blob/0860194ead50c266adec696d0220f41eed5e550e/src/util/RandomGenerator.ts#L15-L16

Reply:
typeorm/src/util/RandomGenerator.ts seems okay to me, no need to install another package, but tbh I would choose a battery tested solution instead